src/js/utils/date-utils.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Create a formatted, locale-aware string representation of a datetime.
+ *
+ * This module contains general purpose functions for localizing and displaying
+ * date-times by timezone, locale and preferred language."
+ *
+ * Most of the heavy lifting is done by the {@link http://momentjs.com/ |'moments' js library}
+ * with the moment-with-locales and moment-timezones add-ons.
+ *
+ *
+ *
+ * Example:
+ *
+ *~~~ javascript
+ * var context = {
+ *    datetime: '2016-10-14 08:00:00+00:00',
+ *    timezone: 'Pacific/Kiritimati',
+ *    language: 'ar',
+ *    format: 'DEFAULT_LONG_DATE_FORMAT'
```

Comment:
This is supposed to decode (is that the right word?) via the getDisplayFormat() method. Am I missing something?